Add campaign requirements (#16676)
sourcegraph/sourcegraph: doc/campaigns/references/requirements.md

# Requirements

> Campaigns has requirements for the Sourcegraph server version, it's connected code hosts and developer environments. 

Reviewer: ```suggestion
Campaigns has requirements for the Sourcegraph server version, its connected code hosts and developer environments. 

```

## Code hosts

> Sourcegraph is compatible with the following code hosts and versions:

Reviewer: Sourcegraph is compatible with more than these code hosts.

```suggestion
Campaigns is compatible with the following code hosts and versions:

```

>   - Github.com
>   - Github Enterprise: version: 2.20 and later
>   - GitLab: 12.7 and later
>     - 13.2 and later includes burndown chart support
>   - Bitbucket server: 5.7+
>   - Notes: 
>       - Webhook support should be enabled to increase performance at scale

Reviewer: ```suggestion
- Github.com
- Github Enterprise 2.20 and later
- GitLab 12.7 and later (burndown charts are only supported with 13.2 and later)
- Bitbucket server 5.7 and later

We **highly recommend** enabling webhooks to increase performance with large campaigns:

- [GitHub](../../admin/external_service/github.md#webhooks)
- [Bitbucket Server](../../admin/external_service/bitbucket_server.md#webhooks)
- [GitLab](../../admin/external_service/gitlab.md#webhooks)

```

> ## Sourcegraph server

Reviewer: "Sourcegraph server" is often used as the informal name for the Docker-container deployment of Sourcegraph (see [docs here](https://docs.sourcegraph.com/admin/install). I think we can reduce it to avoid poossible confusion:

```suggestion
## Sourcegraph
```

Author: 99.9% sure the correct term for managed deployments is now 'Sourcegraph server'. 

Unfortunately, I believe this is incorrect in the handbook. This separate pull request fixes that: https://github.com/sourcegraph/about/pull/2175

> While the latest version of Sourcegraph server is always recommended, version 3.22 or greater is the minimum version required to run campaigns. 

Reviewer: ```suggestion
While the latest version of Sourcegraph server is always recommended, 3.22 is the minimum version required to run campaigns. 
```

"or great" and "minimum version" is redundant

## Requirements for developers creating and running campaigns

>   - Latest version of [src-cli](https://github.com/sourcegraph/src-cli/releases)
>       - Src-cli is supported on Linux or MacOS 
>         - Windows support is experimental
>   - Docker
>   - Git

Reviewer: ```suggestion

- Latest version of the [Sourcegraph CLI `src`](../../cli/index.md)
  - `src` is supported on Linux or macOS , Windows support is experimental
- Docker
- Git
```
